<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>DNA: The Next Generation in Digital Data Storage</title>
    <link rel="stylesheet" href="https://www.w3schools.com/w3css/4/w3.css"> 
    <style>
        .container {
            display: flex; /* Enable flexbox for two-column layout */
        }

        .sidebar {
            width: 300px; /* Adjust width as needed */
            position: sticky;
            top: 0;
            overflow-y: auto; /* Enable scrolling in the sidebar if content overflows */
            height: 100vh;
            padding: 10px
        }

        .content {
            flex: 1; /* Content takes up remaining space */
            overflow-y: auto; /* Enable scrolling for the main content */
            padding: 20px; /* Add padding for visual separation */
        }
        
        .content img {
            max-width: 100%
            height: auto;
        }

        .content pre {
            overflow-x: auto;
            white space: pre-wrap;
            word-wrap: break-word; 
        }

        body {
            font-family: 'Gill Sans', 'Gill Sans MT', Calibri, Trebuchet MS', sans-serif;
        } 

        h1, h2, h3 {
            font-family: Calibri, 'Trebuchet MS', sans-serif;
        }

        /* Existing table styles can remain as they are */
        table {
            font-family: arial, sans-serif;
            border-collapse: collapse;
            width: 100%;
        }

        td, th {
            border: 1px solid #dddddd;
            text-align: left;
            padding: 8px;
        }

        tr:nth-child(even) {
            background-color:   
 #dddddd; 
        }

        p.ex1 {
            margin-left: 250px;
        }

        .myslides {
            display: none;   

        }
    </style>
</head>

<body style="background-color: #FFFFFF; margin-left: 50px;">
    <div class="container">
        <nav class="sidebar">
            <table>
                <tr>
                    <th><h3>Table of Contents</h3></th>
                </tr>
                <tr>
                    <td><a href="#abstract">Abstract</a></td>
                </tr>
                <tr>
                    <td><a href="#introduction">Introduction</a></td>
                </tr>
                <tr>
                    <td><a href="#transcribing">Transcribing Digital to Molecular</a></td>
                </tr>
                <tr>
                    <td><a href="#storage">DNA Storage</a></td>
                </tr>
                <tr>
                    <td><a href="#sequencing">DNA Sequencing</a></td>
                </tr>
                <tr>
                    <td><a href="#references">References</a></td>
                </tr>
            </table>
        </nav>

        <main class="content">
            <h1>ENG0018 Computer Laboratory 2023/24</h1>
            <h2>Student URN: 6825297</h2>
            <hr>
            <h2>DNA, The Next Generation in the Evolution of Digital Data Storage</h2>
            <hr>

            <h3 id="abstract">Abstract</h3>
            <hr>
            <p class="ex1">
                <pre>
Digital data production is growing at unprecedented rates, meaning current methods of data storage are becoming Insufficient to keep up with the amount of data that needs to be stored. 
While still a developing field, a promising solution to this issue is DNA storage. DNA storage achieves a much higher storage density (<a href="#ref3">Buko, T., Tuczko, N. and Ishikawa, T. 2023</a>) 
and has promising potential in terms of durability. Converting digital data into DNA is a process based on binary code
where a specific nucleotide relates to a specific code (00=Adenine,01=cytosine...). The digital data is encoded and translated into a DNA sequence using computer algorithms
and then synthesised. In vivo methods of storage see DNA data storage coming closer to realising its potential for Long-term storage.
To extract digital information from DNA storage, The DNA needs to be sequenced (a process used to identify the sequence of bases in a DNA sequence) and then decoded 
to return the digital data back to its original form.  
                </pre>
            </p>
            <hr>

            <h3 id="introduction">Introduction</h3>
            <hr>
            <p>
                <pre>
Digital Data production has been growing exponentially (<a href="#ref1">Ceze, L., Nivala, J. and Strauss, k., 2019</a>), outpacing growth of the installed base (all devices in active use).
While the total amount of data in 2011 was approximately 5 ZB (5 trillion Gigabytes), it reached 79 ZB in 2021 and continues to display rapid growth.
It is expected to reach 181 ZB in 2025 (<a href="#ref2">Statista research department, 2022</a>). 
                </pre>
            </p>
            <img src="Statista graph.png" alt="Graph showing the exponential growth of digital data" width="700" height="400">
            <p>
                <pre>
Most generated data is discarded, but a fraction still needs to be stored. Due to their cost, space requirements and energy consumption during the recording,
storing and reading of data, current methods of data storage such as magnetic (e.g. hard disk drives), optical (e.g. Blu-ray disks) and solid-state(e.g. flash drives)
are not enough to accommodate such rapid growth (<a href="#ref3">Buko,T., Tuczko, N. and Ishikawa, T. 2023</a>).
Another problem with these methods of data storage is that their durability reaches a maximum of 50 years when in optimal conditions (<a href="#ref4">Grass et al.,2015</a>).
Thus, there is need for a new data storage medium with higher capacity, data density and high durability against extreme conditions. 

Using as few atoms as possible to store one bit of information leads to higher data density,
making molecular data storage attractive (<a href="#ref1">Ceze, L., Nivala, J. and Strauss, k., 2019</a>). Among the many forms of molecular-level data storage,  
DNA is one of the most promising next-generation data carriers. It has a storage density of 1019 bits of data per cubic centimetre,
and its three-dimensional structure makes it approximately eight orders of magnitude denser than other storage media (<a href="#ref3">Buko, T., Tuczko, N. and Ishikawa, T. 2023</a>). 	 
                </pre>
            </p>
            <hr>

            <h3 id="transcribing">Transcribing Digital to Molecular</h3>
            <hr>
            <p>
                <pre>
An article by (<a href="#ref3">Buko, T., Tuczko, N. and Ishakawa, T. 2023</a>) describes that transcribing data into DNA begins with encoding.
Encoding information into a sequence of DNA Is based on binary code, where a nucleotide corresponds to a specific code e.g. 00 = A, 01 = C, 10 = G and 11 = T.  
When binary data is translated into a DNA sequence, it is vital that the creation of long homopolymers, 
DNA polymer whose nucleotide sequence has more than three of the same nucleotides in a row, is avoided,
as well as unreasonable GC content (The percentage of nitrogenous bases in a DNA molecule that are either Guanine or Cytosine), 
as these may lead to mistakes being generated during the synthesis and sequencing (the process by which the nucleotide sequence of a string of DNA is read) of DNA strings.
The process of encoding a file requires converting text into a code such as ASCII,  
A system that is used to convert letters, numbers, symbols, and other text elements into a language that computers understand,
or Base 64 (like ASCII) and then converting the newly coded file into a binary system. 
This process is carried out using a computer algorithm which maps strings of bits (the smallest unit of data that a computer can process and store) into DNA sequences. 
                </pre>
            </p>
            <img src="encoding image.jpg" alt="Table showing how text is converted from digital to DNA format" width="500" height="250">
            <p>
                <pre>
The Resulting DNA sequences are then synthesised, producing many physical copies of each sequence. 
The sequences produced are of varying lengths such that the bit strings are broken into smaller sections' 
which will later need to be reassembled into the original data (<a href="#ref1">Ceze, L., Nivala, J. and Strauss, k., 2019</a>). 
(<a href="#ref5">Heckel, R. et al,2013</a>) found that a simple index-based (the physical location of the DNA sections kept track of) coding scheme is optimal to facilitate reassembly.
Where larger amounts of information need to be stored, a lot of different DNA sequences need to be synthesised. In these scenarios an array-based synthesis is used. 
Array-based synthesis involves binding an array of thousands to millions of nucleic acid fragments to a solid surface
and then bathing the solid surface or “chip” with chosen DNA sequences. 
Complementary base pairing between the added sequences and of DNA and the immobilised nucleic acid fragments 
produces light through fluorescence that can be detected using a specialised machine and used to detect specific DNA sequences. 
This allows many unique sequences to be produced in parallel (<a href="#ref6">Church, G., Gao, Y. and Kosuri, S., 2012</a>) 
                </pre>
            </p>
           
            <hr>
            <h3 id="storage">DNA Storage</h3>
            <hr>
            <p>
                <pre>
Once synthesised the DNA then needs to be stored. 

The most common way to store data within DNA in vitro is solution storage (<a href="#ref3">Buko, T., Tuczko, N. and Ishikawa, T. 2023</a>). 
In earlier attempts, DNA was preserved in ethanol, but as research progressed ammonium-based ionic liquids became more mainstream 
as they improved the stability of the DNA stored through hydrogen bonding between the ionic liquid and the DNA. 
This storage solution, however, only allows DNA to be stored for a year, 
which does not fulfil the goals of DNA data preservation (≥1000 years) (<a href="#ref3">Buko, T., Tuczko, N. and Ishikawa, T. 2023</a>)  
As (<a href="#ref7">Hao, Y. et al.,2020</a>) describe, recently, in vivo (taking place inside a living organism) DNA preservation has been quickly developed. 
The preservation of DNA within a living cell allows the DNA to be replicated with a few orders of magnitude 
during the cell's proliferation processes (The process by which a cell grows and divides to produce two daughter cells), 
much faster than PCR (Polymerase Chain Reaction), A reaction process used to rapidly amplify genes or desired DNA sequences.  
This newer type of DNA storage has potential to “enable both the collection and permanent storage of new data within cellular DNA”  
(<a href="#ref1">Ceze, L., Nivala, J. and Strauss, k., 2019</a>).  
                </pre>
            </p>
            <hr>

            <h3 id="sequencing">DNA Sequencing</h3>
            <hr>
            <p>
                <pre>
To return the DNA sequences back to their digital format they need to undergo  
sequencing (the general laboratory technique for determining the exact sequence of nucleotides in a DNA molecule). 
The most widespread DNA-sequencing platform today is based on image processing  
and the concept of sequencing by synthesis (<a href="#ref1">Ceze, L., Nivala, J. and Strauss, k., 2019</a>). 
In this process, single stranded DNA sequences are attached to a substrate surface and amplified into small physical clusters of copies via PCR, 
and complementary bases with fluorescent markers are then attached one by one to individual sequences. 
The spatial fluorescent pattern is captured in an image and then processed. 
The colour of the fluorescent spots reports the individual bases in the sequence. 
The fluorescent markers are then chemically removed, 
leaving the complementary bases behind and setting up the next base in the sequence to be recognised. 
Once all the bases have been recognised the sequence of bases is then decoded back into the original digital data.  
                </pre>
            </p>
            <hr>
            <h3 id="references">References</h3>
            <hr>
            <pre id="ref1"> 
1.) Ceze, L., Nivala, J. and Strauss, K. (2019) ‘Molecular digital data storage using DNA’, Nature Reviews Genetics, 20(8), pp. 456–466. Available at: https://doi.org/10.1038/s41576-019-0125-3 (Accessed: 15 November 2023) 
            </pre>
            <pre id="ref2">
2.) Petroc, T. Statista Research Department. (2021). 'Volume of data/information created, captured, copied, and consumed worldwide from 2010 to 2020, with forecasts from 2021 to 2025 (in   
 zettabytes)'. Statista. Statista Inc. Accessed: October 30, 2023. https://www.statista.com/statistics/871513/worldwide-data-created/  
            </pre>
            <pre id="ref3">
3.) Buko, T., Tuczko, N. and Ishikawa, T. (2023) ‘DNA Data Storage’, BioTech, 12(2), p. 44. Available at: https://doi.org/10.3390/biotech12020044. (Accessed: November 20, 2023) 
            </pre>
            <pre id="ref4">
4.) Grass, R.N. et al. (2015) ‘Robust Chemical Preservation of Digital Information on DNA in Silica with Error-Correcting Codes’, Angewandte Chemie International Edition,   
 54(8), pp. 2552–2555. Available at: https://doi.org/10.1002/anie.201411378 (Accessed: November 20,2023) 
            </pre>
            <pre id="ref5">
5.) Heckel, R. et al., (2013) ‘Fundamental limits of DNA storage systems’ 2017 IEEE International Symposium on Information Theory (ISIT), Aachen, Germany, 2017, pp. 3130-3134, Available at: 10.1109/ISIT.2017.8007106. (Accessed 21 November,2023)  
            </pre>
            <pre id="ref6">
6.) Church, G., Gao, Y. and Kosuri, S. (2012) ‘Next-Generational Digital Information Storage in DNA’, science, 337, p.1628, Available at: https://www.science.org/doi/10.1126/science.1226355 (Accessed 21 November 2023)
</pre>
            <pre id="ref7">
7.)Hao, Y., Li, Q., Fan, C. and Wang, F. (2020). 'Data Storage Based on DNA. Small Structures'. Available at: https://doi.org/10.1002/sstr.202000046 (accessed december 2023)
            </pre>

        </main>
    </div>
</body>

</html>
 

